app/src/main/java/org/mozilla/rocket/privately/PrivateModeActivity.kt

```diff
@@ -42,6 +55,32 @@ class PrivateModeActivity : BaseActivity(),
     private lateinit var tabViewProvider: PrivateTabViewProvider
     private lateinit var sharedViewModel: SharedViewModel
     private lateinit var screenNavigator: ScreenNavigator
+    private val downloadCompleteReceiver = object : BroadcastReceiver() {
+        override fun onReceive(context: Context, intent: Intent) {
+            val downloadId = intent.getLongExtra(DownloadManager.EXTRA_DOWNLOAD_ID, -1)
+
+            Thread(Runnable {
```

Comment:
make this local broadcastreceiver a general purpose one like
https://github.com/mozilla-tw/FirefoxLite/commit/cca8521f9ede4c3c3de8a00f7f748db4d3c43848

use ThreadUtil for bg thread work, and then add comment to replace this with lifecycle aware coroutine.